src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -211,10 +211,31 @@ public void resetStudentGoogleId(String originalEmail, String courseId)
                             .withGoogleId(null)
                             .build());
         } catch (InvalidParametersException | EntityAlreadyExistsException e) {
-            Assumption.fail("Resting google ID shall not cause: " + e.getMessage());
+            Assumption.fail("Resetting google ID shall not cause: " + e.getMessage());
         }
     }
 
+    /**
+     * Regenerates the registration key for the student represented by {@code originalStudent}.
+     * @return Returns the student attributes with the new registration key.
+     */
+    public StudentAttributes regenerateStudentRegistrationKey(StudentAttributes originalStudent) {
+        try {
+            studentsDb.deleteStudent(originalStudent.getCourse(), originalStudent.getEmail());
+            StudentAttributes updatedStudent;
+
+            do {
```

Comment:
I don't think it is a good idea to introduce a loop here :P. Maybe just throw exception when the generated key is the same?